src/lightgbm/src/main/scala/LightGBMUtils.scala

```diff
@@ -133,7 +133,13 @@ object LightGBMUtils {
   def getNumCoresPerExecutor(dataset: Dataset[_]): Int = {
     val spark = dataset.sparkSession
     import spark.implicits._
-    spark.range(0, 1).map(_ => java.lang.Runtime.getRuntime.availableProcessors).collect.head
+    val numMachineCores = spark.range(0, 1)
```

Comment:
Looks good! @imatiach-msft is it possible to remove the numMachineCores calculation and just use the confCores calc?

Reply:
updated based on our discussion (put the conf code inside a try/catch)